src/com/projectkorra/projectkorra/GeneralMethods.java

```diff
@@ -978,6 +978,31 @@ public static double getDistanceFromLine(final Vector line, final Location point
 		return new ArrayList<>(location.getWorld().getNearbyEntities(location, radius, radius, radius, entity -> !(entity.isDead() || (entity instanceof Player && ((Player) entity).getGameMode().equals(GameMode.SPECTATOR)))));
 	}
 
+	/**
+	 * Gets a {@code List<Entity>} of entities around a specified radius from
+	 * the specified area while ignoring entities that may be behind walls.
+	 *
+	 * @param location The base location
+	 * @param radius The radius of blocks to look for entities from the location
+	 * @param respectBlocks Whether or not to check for any blocks between the location and entity.
+	 * @return A list of entities around a point
+	 */
+	public static List<Entity> getEntitiesAroundPoint(final Location location, final double radius, final boolean respectBlocks) {
+		List<Entity> validEntities = getEntitiesAroundPoint(location, radius);
+
+		if (respectBlocks) {
+			for (Entity entity : validEntities) {
+				Location entityLocation = entity.getLocation();
+				Vector direction = new Vector(1, 0, 0).add(entityLocation.toVector().subtract(location.toVector()));

```

Comment:
Can you explain the vector math here?

Reply:
just moves the location from point a to point b (being from the stream to the detected entity) and checks for blocks along the way

Reply:
Yeah I got that much, but can you explain each part? Like why you have that initial vector going in the positive x direction?

Reply:
if you have a better way to detect the blocks between the 2 points i’d be open to suggestions

Reply:
GeneralMethods#getBlocksAlongLine

Reply:
Not the best tbh but you could use it as a reference

Reply:
![image](https://user-images.githubusercontent.com/19560788/73616051-845f5180-45c3-11ea-82db-f754615e9706.png)

This vector math will not work in every situation. Take my hypothetical scenario where you have vector **b** as the entity's location, vector **a** as the location passed in, your vector math gives us the vector **u** Once we add U back to the initial location a we get **b+i** and if you look in my picture, you'll see that in the hypothetical scenario where there are actual blocks between a and b, your code would take it as no blocks in between.

The best way would be to simply check if the vector between the two locations intersects with a block or has a block material other than air.